pkgs/tools/misc/ttmkfdir/default.nix

```
@@ -20,11 +20,23 @@ stdenv.mkDerivation {
       ./cstring.patch # also fixes some other compilation issues (freetype includes)
     ];
 
-  preInstall = ''
-    mkdir -p $out; makeFlags="DESTDIR=$out BINDIR=/bin"
+  # cross-compilation fixes:
+  # - fix libtool, the reason it does not work in nativeBuildInputs is complicated
+  #   see https://github.com/NixOS/nixpkgs/pull/192878 for more info
+  # - don't hardcode CXX, respect the value from the environment
+  # - freetype-config doesn't properly support cross-compilation, but is just a thin
+  #   wrapper around pkg-config anyways
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace "libtool " "${libtool}/bin/libtool " \
+      --replace "CXX=" "#CXX=" \
+      --replace "freetype-config" "${stdenv.cc.targetPrefix}pkg-config freetype2"
   '';
 
-  buildInputs = [freetype fontconfig libunwind libtool flex bison];
+  makeFlags = [ "DESTDIR=$out" "BINDIR=/bin" ];
```

How about this as an alternative to patching CXX?
```suggestion
  makeFlags = [ "DESTDIR=$out" "BINDIR=/bin" "CXX=${stdenv.cc.targetPrefix}c++" ];
```

---

good point! that works too